### ISSUE TITLE
Redshift: parsing error when some parameters present in copy statement

### ISSUE THREAD

**Issue by Clepech** [sqlfluff/sqlfluff#3152]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Linter fail to parse when some parameters are present in [copy command](https://docs.aws.amazon.com/redshift/latest/dg/r_COPY.html). The error was observed with copy csv parameters `IGNOREHEADER` and `ACCEPTINVCHARS` but might occur with other parameters

### Expected Behaviour

Linter should success

### Observed Behaviour

Linter fails to parse

```
== [foo.sql] FAIL                                                                                                                                                               
L:   4 | P:  30 |  PRS | Line 4, Position 30: Found unparsable section:
                       | '\nIGNOREHEADER 1\nACCEPTINVCHARS'
```

### How to reproduce

Run lint command:

```cmd
sqlfluff lint --dialect=redshift foo.sql
```

with file `foo.sql`:

```sql
COPY some_table
FROM 's3://some_bucket/some/path'
IAM_ROLE 'some_iam_role'
FORMAT AS CSV TRUNCATECOLUMNS
IGNOREHEADER 1
ACCEPTINVCHARS
;

```

### Dialect

Redshift

### Version

0.12.0

### Configuration

[tool.sqlfluff.core]

encoding = "utf8"
# Rules details: https://docs.sqlfluff.com/en/stable/rules.html
rules = "L001,L002,L003"

[tool.sqlfluff.indentation]
indented_joins = false
template_blocks_indent = false

[tool.sqlfluff.rules]
max_line_length = 120
allow_scalar = false
comma_style = "trailing"

[tool.sqlfluff.rules.L010]  # Keywords
# L020 & L030 inherit from this rule
capitalisation_policy = "upper"

[tool.sqlfluff.templater.jinja]
library_path="mock_tempates"

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
